cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -158,12 +163,23 @@ newtype ByronUpdateState = ByronUpdateState Byron.Update.State
 
 newtype UTxO era = UTxO (Map TxIn (TxOut era))
 
+instance IsCardanoEra era => ToJSON (UTxO era) where
+  toJSON (UTxO m) = Aeson.Array . Vector.fromList . map convert $ Map.toList m
+   where
+    convert :: (TxIn, TxOut era) -> Aeson.Value
+    convert ((TxIn txId (TxIx ix), txout)) =
+      let txin = ( Text.decodeUtf8 (serialiseToRawBytesHex txId)
+                 <> "#"
+                 <> Text.pack (show ix)
+                 )
```

Comment:
Yes, but this is what we should have put in the `ToJSON` instance for `TxIn`! Then we could use a more general instance here.